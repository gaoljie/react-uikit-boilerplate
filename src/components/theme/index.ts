import React from "react";

export const theme = {
  primary: "#EF6F35",
  primaryFont: "Montserrat",
  secondaryFont: "Work Sans",
  desktop: "@media (max-width: 1920px)",
  smallDesktop: "@media (max-width: 1441px)",
  tablet: "@media (max-width: 769px)",
  mobile: "@media (max-width: 600px)"
};

export const ThemeContext = React.createContext(theme);

export const ThemeProvider = ThemeContext.Provider;
