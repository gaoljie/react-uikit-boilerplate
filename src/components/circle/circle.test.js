import React from "react";
import "jest-styled-components";
import { mount } from "enzyme";
import theme from "../theme";

import Circle from "./index";

const props = {
  color: "black",
  width: "3px"
};

describe("Test Circle", () => {
  it("Matches the snapshot", () => {
    const wrapper = mount(<Circle {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Styling", () => {
    const wrapper = mount(<Circle />);

    expect(wrapper).toHaveStyleRule("width", "2px");
    expect(wrapper).toHaveStyleRule("height", "2px");
    expect(wrapper).toHaveStyleRule("background-color", theme.primary);

    wrapper.setProps({ width: "3px", color: "black" });

    expect(wrapper).toHaveStyleRule("width", "3px");
    expect(wrapper).toHaveStyleRule("height", "3px");
    expect(wrapper).toHaveStyleRule("background-color", "black");

    wrapper.setProps({ half: true });

    expect(wrapper).toHaveStyleRule("background-color", "white");
  });
});
