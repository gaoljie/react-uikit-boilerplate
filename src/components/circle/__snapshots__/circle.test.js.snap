// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Circle Matches the snapshot 1`] = `
.c0 {
  width: 3px;
  color: black;
  width: 3px;
  height: 3px;
  display: inline-block;
  border: 1px solid black;
  background-color: black;
  border-radius: 50%;
}

<Styled(styled.div)
  color="black"
  width="3px"
>
  <StyledComponent
    color="black"
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  ",
            [Function],
            "
",
            "
  width: ",
            [Function],
            ";
  height: ",
            [Function],
            ";
  display: inline-block;
  border: 1px solid ",
            [Function],
            ";
  background-color: ",
            [Function],
            ";
  border-radius: 50%;
",
          ],
        },
        "displayName": "Styled(styled.div)",
        "foldedComponentIds": Array [
          "sc-bdVaJa",
        ],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    width="3px"
  >
    <div
      className="c0"
      color="black"
      width="3px"
    />
  </StyledComponent>
</Styled(styled.div)>
`;
